### PR TITLE
Sort and log proxy selection

### PR DESCRIPTION
**Pull request by jsantos17**
[ch10206]

### CODE REVIEW

**Comment by jsantos17** [comment]
Ideally we could simply enable the logging inside `proxy-vole`. But the library's logging is broken.

cc @beckyconning @djspiewak

File: datasource/src/main/scala/quasar/physical/s3/AsyncHttpClientBuilder.scala [hunk 1]
```diff
@@ -52,16 +56,31 @@ object AsyncHttpClientBuilder extends Http4sClientBuilder {
         s"http4s-async-http-client-worker-${i}"
       })).build()
 
+  private[s3] def sortProxies(proxies: List[Proxy]): List[Proxy] =
+    proxies.sortWith((l, r) => (l.`type`, r.`type`) match {
+      case (ProxyType.HTTP, ProxyType.DIRECT) => true
+      case (ProxyType.SOCKS, ProxyType.DIRECT) => true
+      case _ => false
```
Comment (rintcius) [inline datasource/src/main/scala/quasar/physical/s3/AsyncHttpClientBuilder.scala]: I think there has to be one more case set to true according to the contract? Either HTTP < SOCKS or the other way around. 

I don't mind sorting that way in this case, but usually I prefer to map to int and then use `sortBy` - that scales better when the amount of elements grows.

Comment (jsantos17) [reply datasource/src/main/scala/quasar/physical/s3/AsyncHttpClientBuilder.scala]: >I think there has to be one more case set to true according to the contract?

Not necessarily. [`sortWith`'s documentation indicates it implements a stable sort](https://www.scala-lang.org/api/current/scala/collection/immutable/List.html#sortWith(lt:(A,A)=>Boolean):C), so elements deemed "equal" will appear in the same order in the resulting list. I'd rather not modify the order `SOCKS` and `HTTP`.

Comment (rintcius) [reply datasource/src/main/scala/quasar/physical/s3/AsyncHttpClientBuilder.scala]: Ah yes, sorry - you are right.

**Pull request merged**